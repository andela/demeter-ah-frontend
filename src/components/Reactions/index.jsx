import React from 'react';
import comment from '../../assets/images/article/comment.png';
import LikeIcon from '../../assets/svgs/likeIcon';
import DisLikeIcon from '../../assets/svgs/disLike';
import Bookmark from '../../assets/svgs/bookmarksm';
import './index.scss';

const Reactions = ({
  isAuthenticated, articleAuthorUsername, isBookmarked, authUsername,
  viewComment, commentNo, bookmarkArticle, voteAction,
  article: { upVote = [], downVote = [], slug } = {}, user
}) => {
  const notbookmarked = (<Bookmark onClick={bookmarkArticle} isBookmarked slug={slug} />);
  const bookmarked = (
    <span className="bookmarkIconFill">
      {notbookmarked}
    </span>
  );

  const isLiked = [...upVote, ...downVote].find(
    vote => vote.user.username === user.username
  ) || { status: null };

  const [like, setLike] = React.useState(false);
  const [dislike, setDislike] = React.useState(false);
  const [isVoting, setVoting] = React.useState(false);

  const handleLike = async (status) => {
    setVoting(true);
    setLike(!like);
    setDislike(false);
    await voteAction({ status, slug });
    setVoting(false);
  };

  const handleDisLike = async (status) => {
    setVoting(true);
    setDislike(!dislike);
    setLike(false);
    await voteAction({ status, slug });
    setVoting(false);
  };

  return (
    <div className="action share-section">
      <div className="action-con">
        <div className="like-section">
          <button
            disabled={isVoting}
            onClick={
              () => handleLike(true)
            }
            className={`icons like-icon ${(isVoting ? like : isLiked.status === true) ? 'active' : ''}`}
          >
            <LikeIcon />
          </button>
          <p className="like-count">{upVote.length}</p>
        </div>
        <div className="dislike-section">
          <button
            disabled={isVoting}
            onClick={
              () => handleDisLike(false)
            }
            className={`icons dislike-icon ${(isVoting ? dislike : isLiked.status === false) ? 'active' : ''}`}
          >
            <DisLikeIcon />
          </button>
          <p className="dislike-count">{downVote.length}</p>
        </div>
        <div className="comment-section">
          <button onClick={viewComment} className="comment-icon icons">
            <img
              src={comment}
              className="comment-img"
              alt="Comment"
            />
          </button>
          <p className="comment-count">{commentNo}</p>
        </div>
        <div className="bookmark-section">
          <button className="bookmark-icon">
            {isAuthenticated && (articleAuthorUsername !== authUsername) && isBookmarked ? bookmarked : ''}
            {isAuthenticated && (articleAuthorUsername !== authUsername) && !isBookmarked ? notbookmarked : ''}
          </button>
        </div>
      </div>
    </div>
  );
};

export default Reactions;
