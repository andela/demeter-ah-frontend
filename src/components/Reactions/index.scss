.action {
  .bookmarkIconFill {
    path {
      fill: #835BD8;
    }
  }
}
