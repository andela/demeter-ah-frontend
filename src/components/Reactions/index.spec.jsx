import React from 'react';
import { shallow } from 'enzyme';
import Reactions from './index';

describe('Reactions Component', () => {
  it('Should render without errors', () => {
    const component = shallow(<Reactions />);
    expect(component).toMatchSnapshot();
  });
});
