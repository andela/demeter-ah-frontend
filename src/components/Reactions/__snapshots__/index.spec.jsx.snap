// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reactions Component Should render without errors 1`] = `
<div
  className="action share-section"
>
  <div
    className="action-con"
  >
    <div
      className="like-section"
    >
      <button
        className="icons like-icon "
        disabled={false}
        onClick={[Function]}
      >
        <likeIcon />
      </button>
      <p
        className="like-count"
      >
        0
      </p>
    </div>
    <div
      className="dislike-section"
    >
      <button
        className="icons dislike-icon "
        disabled={false}
        onClick={[Function]}
      >
        <disLike />
      </button>
      <p
        className="dislike-count"
      >
        0
      </p>
    </div>
    <div
      className="comment-section"
    >
      <button
        className="comment-icon icons"
      >
        <img
          alt="Comment"
          className="comment-img"
          src="/article/comment.png"
        />
      </button>
      <p
        className="comment-count"
      />
    </div>
    <div
      className="bookmark-section"
    >
      <button
        className="bookmark-icon"
      />
    </div>
  </div>
</div>
`;
